File: Python/TextConverter/text_converter.py

# This is for Python 3
import html as html_converter

# for Python 2 uncomment this line
#import cgi as html_converter

class UnicodeFileToHtmlTextConverter(object):

    def __init__(self, full_filename_with_path):
        self.full_filename_with_path = full_filename_with_path

    def convert_to_html(self):
        f = open(self.full_filename_with_path, "r")
        html = ""
        for line in f:
            line = line.rstrip()
            html += html_converter.escape(line, quote=True)
            html += "<br />"

        return html

File: Ruby/TextConverter/text_converter.rb
require 'cgi'

class UnicodeFileToHtmlTextConverter
  attr_reader :full_path_to_file
  
  def initialize(full_path_to_file)
    @full_path_to_file = full_path_to_file
  end

  def convert_to_html
    f = File.open(@full_path_to_file)
    html = ""
    f.each_line do |line|
      line = line.strip
      html << CGI.escapeHTML(line)
      html << "<br />"
    end

    return html
  end
end
